ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -1643,12 +1746,25 @@ private boolean resolveRange(long totalSize) {
       return failure;
     }
 
+    /**
+     * Wrapper over resolveRange
+     * @param dataRangeSize byte range to be used for range resolution
+     * @return {@code true} if range resolution succeeded. {@code false} otherwise
+     */
+    protected boolean resolveRangeProper(long dataRangeSize) {
+      /**
+       * resolveRange is a weird function that returns false on success and true on failure !
+       * Hence, this wrapper to make code readable !
+       */
+      return !resolveRange(dataRangeSize);
+    }
```

Comment:
or we should just change the return value in resolveRange method.

Reply:
Yeah i thought of it and its going to break a lot of code

Reply:
I can fix existing code in another PR